test/upload-download.test.js

```diff
@@ -91,6 +91,7 @@ describe('storage service', function() {
       .set('Content-Type', 'application/json')
       .expect('Content-Type', /json/)
       .expect(200, function(err, res) {
+        if (err) done(err);
```

Comment:
Shouldn't this be:
```
if (err) return done(err);
```
Can you clarify what you are not returning here.

The same applies for other places.